vllm/model_executor/custom_op.py

```diff
@@ -138,15 +156,29 @@ def default_on() -> bool:
     # - MyOp.enabled()
     # - op_registry["my_op"].enabled()
     op_registry: dict[str, type['CustomOp']] = {}
+    op_registry_oot: dict[str, type['CustomOp']] = {}
 
     # Decorator to register custom ops.
+    # For OOT custom ops:
+    #   if in-tree layer class is registered with an oot_custom_op layer,
+    #   the oot_custom_op layer will be used instead.
+    #   use `name=<In-Tree Layer class Name>` and `is_oot_custom_op=True`
+    # Examples:
+    # - @CustomOp.register("UnquantizedFusedMoEMethod", is_oot_custom_op=True)
+    # - @CustomOp.register("RMSNorm", is_oot_custom_op=True)
     @classmethod
-    def register(cls, name: str):
+    def register(cls, name: str, is_oot_custom_op=False):
```

Comment:
I want to bikeshed the names. We shouldn't use abbreviations ("oot" is hard to read), instead we should expand them to their full names.

Reply:
Gotcha, I see that we decided to do this half a year ago, so let's let that stick.